src/stores/firebaseAuthStore.ts

```diff
@@ -367,6 +386,75 @@ export const useFirebaseAuthStore = defineStore('firebaseAuth', () => {
     return response.json()
   }
 
+  /**
+   * Fetch the current cloud subscription status for the authenticated user
+   * @returns Subscription status or null if no subscription exists
+   */
+  const fetchSubscriptionStatus =
```

Comment:
The subscription logic might be better of in it's own dedicated store.

Reply:
@jtydhr88 we can't put this in a subscription store that we can dynamically import?